components/director/internal/domain/runtime/service.go

```diff
@@ -608,26 +617,30 @@ func extractUnProtectedLabels(labels map[string]*model.Label, protectedLabelsKey
 	return result, nil
 }
 
-func unsafeExtractUnProtectedLabels(labels map[string]interface{}, protectedLabelsKeyPattern string) (map[string]interface{}, error) {
+func unsafeExtractModifiableLabels(labels map[string]interface{}, protectedLabelsKeyPattern string, immutableLabelsKeyPattern string) (map[string]interface{}, error) {
 	result := make(map[string]interface{})
 	for labelKey, label := range labels {
-		protected, err := isProtected(labelKey, protectedLabelsKeyPattern)
+		modifiable, err := isLabelModifiable(labelKey, protectedLabelsKeyPattern, immutableLabelsKeyPattern)
 		if err != nil {
-			return nil, err
+			return result, err
 		}
-		if !protected {
+		if modifiable {
 			result[labelKey] = label
 		}
 	}
 	return result, nil
 }
 
-func isProtected(labelKey string, labelKeyPattern string) (bool, error) {
-	matched, err := regexp.MatchString(labelKeyPattern, labelKey)
+func isLabelModifiable(labelKey, protectedLabelsKeyPattern, immutableLabelsKeyPattern string) (bool, error) {
+	protected, err := regexp.MatchString(protectedLabelsKeyPattern, labelKey)
+	if err != nil {
+		return false, err
+	}
+	immutable, err := regexp.MatchString(immutableLabelsKeyPattern, labelKey)
 	if err != nil {
 		return false, err
 	}
-	return matched, nil
+	return !protected && !immutable, err
```

Comment:
About this function we have two checks/conditions and my concert is if someone pass a string label that match both of them because the "immutable" pattern is just `xsappnameCMPClone`. For example - `someTextxsappnameCMPCloneAnotherText_defaultEventing`. I assume it will match and in this case we will return error?

Reply:
It will match because of the "_defaultEventing" suffix

Reply:
Yep, that's the "protected" label pattern but it will also match the "immutable" pattern which is `xsappnameCMPClone`